design-principles.md

```diff
@@ -273,13 +277,29 @@ API design
   - String manipulation to generate data or code that will subsequently
     be interpreted or parsed is a bad practice as it leads to code
     injection vulnerabilities.
+    
+  - APIs should not hide expensive operations. 
 
-* If we wrap some aspect of a service from one environment for exposure
-  in another environment (for example, exposing an Android API in Dart),
-  we should expose/wrap all of it, so that there's no cognitive cliff
+* Convenience APIs that wrap some aspect of a service from one environment
+  for exposure in another environment (for example, exposing an Android API
+  in Dart), should expose/wrap the complete API, so that there's no cognitive cliff
   when interacting with that service (where you are fine using the exposed
   API up to a point, but beyond that have to learn all about the underlying
   service).
+  
+* APIs that wrap underlying services but prevent the underlying API from
```

Comment:
Long sentence fatigue.  Maybe "... parts of the underlying API.  This may require refactoring features so that they are more usable, and ..."